diff -Naur AstroMenaceSourceCode-orig/AstroMenaceSource/ScriptEngine/Setup.cpp AstroMenaceSourceCode/AstroMenaceSource/ScriptEngine/Setup.cpp
--- AstroMenaceSourceCode-orig/AstroMenaceSource/ScriptEngine/Setup.cpp	2007-09-30 13:19:27.000000000 +0200
+++ AstroMenaceSourceCode/AstroMenaceSource/ScriptEngine/Setup.cpp	2007-12-06 22:30:37.000000000 +0100
@@ -48,9 +48,9 @@
 	Setup.BuildVersion = GAME_VERSION_BUILD;
 
 
-	Setup.Width = CurrentVideoMode.W;
-	Setup.Height = CurrentVideoMode.H;
-	Setup.BPP = CurrentVideoMode.BPP;
+	Setup.Width = 640;
+	Setup.Height = 480;
+	Setup.BPP = 0;
 	if ((Setup.Width*1.0f)/(Setup.Height*1.0f) < 1.4f)
 	{
 		Setup.fAspectRatioWidth = 1024.0f;
@@ -586,9 +586,9 @@
 		}
 		if (ResolutionNeedReset)
 		{
-			Setup.Width = CurrentVideoMode.W;
-			Setup.Height = CurrentVideoMode.H;
-			Setup.BPP = CurrentVideoMode.BPP;
+			Setup.Width = 640;
+			Setup.Height = 480;
+			Setup.BPP = 0;
 		}
 	}
 
