diff -urN AstroMenace/AstroMenaceSource/config.h AstroMenace-patched/AstroMenaceSource/config.h
--- AstroMenace/AstroMenaceSource/config.h	2012-10-21 22:45:10.000000000 +1100
+++ AstroMenace-patched/AstroMenaceSource/config.h	2012-10-23 02:46:58.884054328 +1100
@@ -64,7 +64,7 @@
 // #define compression
 
 // define DATADIR with gamedata.vfs file location if you don't put it together in the same directory with binary file
-// #define DATADIR "/usr/share/astromenace"
+#define DATADIR "/usr/share/astromenace"
 
 
 
