diff -urN AstroMenaceSourceCode/AstroMenaceSource/Core/Sound/SoundManager.cpp AstroMenaceSourceCode-patched/AstroMenaceSource/Core/Sound/SoundManager.cpp
--- AstroMenaceSourceCode/AstroMenaceSource/Core/Sound/SoundManager.cpp	2007-09-30 21:45:51.000000000 +1100
+++ AstroMenaceSourceCode-patched/AstroMenaceSource/Core/Sound/SoundManager.cpp	2009-11-21 23:09:59.160776272 +1000
@@ -67,7 +67,7 @@
 	{
 		strcat(Err, alcGetString(Device, ErrCode));
 		strcat(Err, "\n");
-		fprintf(stderr, Err);
+		fprintf(stderr, "%s", Err);
 		return AL_FALSE;
 	}
 	return AL_TRUE;
@@ -82,7 +82,7 @@
 	{
 		strcat(Err, alGetString(ErrCode));
 		strcat(Err, "\n");
-		fprintf(stderr, Err);
+		fprintf(stderr, "%s", Err);
 		return AL_FALSE;
 	}
 	return AL_TRUE;
@@ -97,7 +97,7 @@
 	{
 		strcat(Err, alutGetErrorString(ErrCode));
 		strcat(Err, "\n");
-		fprintf(stderr, Err);
+		fprintf(stderr, "%s", Err);
 		return AL_FALSE;
 	}
 	return AL_TRUE;
